src/Validator.cc

```diff
@@ -7,6 +7,7 @@
 #include "Validator.h"
 
 Validator::Result Validator::validate(ChcDirectedHyperGraph const & graph, VerificationResult const & result) {
+    if (not result.hasWitness()) { return Validator::Result::NOT_VALIDATED; }
```

Comment:
That's for some of the cases, when we shrink graph to single TS?

Reply:
Same question about `noWitness` structure

Reply:
There are other situations where we might not be able to produce the witness at the moment.